ui/src/components/SampleView/ContextMenu.jsx

```diff
@@ -314,19 +339,23 @@ export default function ContextMenu(props) {
         ...(enable2DPoints
           ? [
               { text: 'divider', key: 4 },
-              {
-                text: 'Data Collection (Limited OSC)',
-                action: () => createPointAndShowModal('DataCollection'),
-                key: 5,
-              },
-              {
-                text: 'Characterisation (1 Image)',
-                action: () =>
-                  createPointAndShowModal('Characterisation', {
-                    num_imags: 1,
-                  }),
-                key: 6,
-              },
+              availableMethods.has('datacollection')
+                ? {
+                    text: 'Data Collection (Limited OSC)',
+                    action: () => createPointAndShowModal('DataCollection'),
+                    key: 5,
```

Comment:
The filtering loop below relied on the `key` property to hide options. Since this object's key was not called `'datacollection'`, the option would appear even if `datacollection` was set to `false` in `available_methods`.

Same with the "Characterisation (1 Image)" option below.